### PR TITLE
Bump to Alpine 3.20, Fedora 40, FreeBSD 13.4 and 14.1

### PR DESCRIPTION
**Pull request by clementperon**

### CODE REVIEW

File: Pcap++/src/PcapLiveDevice.cpp [hunk 1]
```diff
@@ -1011,15 +1011,15 @@ namespace pcpp
...
-		if (sysctl(mib, 6, buf, &len, nullptr, 0) < 0)
+		if (sysctl(mib, 6, buf.data(), &len, nullptr, 0) < 0)
 		{
 			PCPP_LOG_DEBUG("Error in retrieving MAC address: sysctl 2 error");
 			return;
 		}
 
-		struct if_msghdr* ifm = (struct if_msghdr*)buf;
+		struct if_msghdr* ifm = (struct if_msghdr*)buf.data();
 		struct sockaddr_dl* sdl = (struct sockaddr_dl*)(ifm + 1);
```
Comment (Dimi1010) [inline Pcap++/src/PcapLiveDevice.cpp:1023]: Do we actually need the `struct` keywords here? Also possibly switch the casts to reinterpret casts?

File: Pcap++/src/PcapLiveDevice.cpp [hunk 2]
```diff
@@ -1011,15 +1011,15 @@ namespace pcpp
 			return;
 		}
 
-		uint8_t buf[len];
+		std::vector<uint8_t> buf(len);
 
-		if (sysctl(mib, 6, buf, &len, nullptr, 0) < 0)
+		if (sysctl(mib, 6, buf.data(), &len, nullptr, 0) < 0)
```
Comment (Dimi1010) [inline Pcap++/src/PcapLiveDevice.cpp:1016]: Would probably be more correct for the length to be taken from `buf.size()`.

File: Pcap++/src/PcapLiveDevice.cpp [hunk 3]
```diff
@@ -1011,15 +1011,15 @@ namespace pcpp
 			return;
 		}
 
-		uint8_t buf[len];
+		std::vector<uint8_t> buf(len);
```
Comment (Dimi1010) [inline Pcap++/src/PcapLiveDevice.cpp:1014]: Tbh, since `len` is a non-constexpr variable I have... questions about how this compiled before...

**Review by Dimi1010 — APPROVED** [review]
LGTM. Made a couple of comments on the changes in the cpp, but they aren't critical.

File: Pcap++/src/PcapLiveDevice.cpp (hunk 3 above)
Comment (clementperon) [reply Pcap++/src/PcapLiveDevice.cpp:1014]: This is a VLA allocated on stack

File: Pcap++/src/PcapLiveDevice.cpp (hunk 1 above)
Comment (clementperon) [reply Pcap++/src/PcapLiveDevice.cpp:1023]: I don't think so, but this commit is only their to fix VLA build issue with new FreeBSD version, not to clean this part of code.

**Review by tigercosmos — APPROVED** [review]
LGTM, but maybe we should do the upgrade before the release though. 😂

**Comment by clementperon** [comment]
> LGTM, but maybe we should do the upgrade before the release though. 😂

Yes we could have done it sooner, but priority was Ubuntu 2

**Review by seladb — APPROVED** [review]

**Pull request merged**